Source/Private/ASCollectionLayout.mm

```diff
@@ -175,4 +206,123 @@ - (CGSize)viewportSize
   }
 }
 
+# pragma mark - Convenient inline functions
+
+ASDISPLAYNODE_INLINE ASSizeRange ASCollectionLayoutElementSizeRangeFromSize(CGSize size)
+{
+  // The layout delegate consulted us that this element must fit within this size,
+  // and the only way to achieve that without asking it again is to use an exact size range here.
+  return ASSizeRangeMake(size);
+}
+
+ASDISPLAYNODE_INLINE void ASCollectionLayoutSetSizeToElement(CGSize size, ASCollectionElement *element)
+{
+  if (ASCellNode *node = element.node) {
+    if (! CGSizeEqualToSize(size, node.frame.size)) {
+      CGRect frame = CGRectZero;
+      frame.size = size;
+      node.frame = frame;
+    }
+  }
+}
+
+/**
+ * Measures all elements in the specified rect and blocks the calling thread while measuring those in the blocking rect.
+ */
+ASDISPLAYNODE_INLINE void ASCollectionLayoutMeasureElementsInRects(CGRect rect, CGRect blockingRect, ASCollectionLayoutState *layout)
```

Comment:
There's too much code in this function IMO for it to be inlined. I'd make it a plain old C function or a class method.